### PR TITLE
Improve & Move "How to contribute" section to Top level readme 

### PR DESCRIPTION
**Pull request by AdarshRawat1**
## Reference 
Based on Discussion in 
- https://github.com/p4lang/p4c/pull/4573

## Task  
- [x] Moved "How to contribute" section to Top level readme 
- [ ] Added Communication channels for new contributors
- [x] Link Guidelines 

## Reason  
- Better organization

### CODE REVIEW

**Comment by fruffy** [comment]
There is some duplication going on between this and `CONTRIBUTING.md`. We should just direct to CONTRIBUTING now that we have added it.

**Comment by AdarshRawat1** [comment]
I agree sir ,
that is the reason why I marked this PR as draft.

I am on it 😀.

File: README.md [hunk 1]
```diff
@@ -626,3 +626,7 @@ access them from the IR
 * Tables with multiple apply calls
 
 See also [unsupported P4_16 language features](backends/bmv2/README.md#unsupported-p4_16-language-features).
+
+# ⭐ How to contribute
```
Comment (fruffy) [inline README.md:630]: The star looks strange in the README, it is very large and only used here. I would remove it.

Comment (AdarshRawat1) [reply README.md:630]: Sure !

**Review by fruffy — APPROVED** [review]

**Comment by fruffy** [comment]
You can mark this PR ready and I will add it to the merge queue: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue

**Review by fruffy — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -626,3 +626,29 @@ access them from the IR
 * Tables with multiple apply calls
 
 See also [unsupported P4_16 language features](backends/bmv2/README.md#unsupported-p4_16-language-features).
+
+# How to contribute
```
Comment (fruffy) [inline README.md:630]:
```suggestion
# How to Contribute
```

File: README.md [hunk 3]
```diff
@@ -626,3 +626,29 @@ access them from the IR
...
+For contributing to P4 Compiler, it’s good to know some P4. Here are some learning materials:
+
+- General hands-on [tutorials](https://github.com/p4lang/tutorials).
+- [Technical documentation on P4-related topics](https://github.com/jafingerhut/p4-guide?tab=readme-ov-file#introduction).
+- Motivating P4: [IEEE ICC 2018 // Keynote: Nick McKeown, Programmable Forwarding Planes Are Here To Stay](https://www.youtube.com/watch?v=8ie0FcsN07U)
+- Introducing P4-16 in detail: 
+  - Part 1: [Introduction to P4_16. Part 1](https://www.youtube.com/watch?v=GslseT4hY1w)
+  - Part 2: [Introduction to P4_16. Part 2](https://www.youtube.com/watch?v=yqxpypXIOtQ)
+- Material on the official P4 compiler: 
+  - [Understanding the Open-Soure P416 Compiler - February 15, 2022 - Mihai Budiu](https://www.youtube.com/watch?v=Rx5AQ0IF6eU)
```
Comment (fruffy) [inline README.md:644]:
```suggestion
  - [Understanding the Open-Source P4-16 Compiler - February 15, 2022 - Mihai Budiu](https://www.youtube.com/watch?v=Rx5AQ0IF6eU)
```

File: README.md [hunk 4]
```diff
@@ -626,3 +626,29 @@ access them from the IR
 * Tables with multiple apply calls
 
 See also [unsupported P4_16 language features](backends/bmv2/README.md#unsupported-p4_16-language-features).
+
+# How to contribute
+
+We welcome and appreciate new contributions. Please take a moment to review our [Contribution Guidelines](CONTRIBUTING.md) to get started.
+
+# P4 Compiler Onboarding
+For contributing to P4 Compiler, it’s good to know some P4. Here are some learning materials:
```
Comment (fruffy) [inline README.md:635]:
```suggestion
Educational material on P4:
```

**Review by fruffy — CHANGES_REQUESTED** [review]

**Review by fruffy — APPROVED** [review]

**Pull request merged**